dbt/adapters/snowflake/connections.py

```diff
@@ -218,38 +218,48 @@ def open(cls, connection):
             logger.debug('Connection is already open, skipping open.')
             return connection
 
-        try:
-            creds = connection.credentials
-
-            handle = snowflake.connector.connect(
-                account=creds.account,
-                user=creds.user,
-                database=creds.database,
-                schema=creds.schema,
-                warehouse=creds.warehouse,
-                role=creds.role,
-                autocommit=True,
-                client_session_keep_alive=creds.client_session_keep_alive,
-                application='dbt',
-                **creds.auth_args()
-            )
-
-            if creds.query_tag:
-                handle.cursor().execute(
-                    ("alter session set query_tag = '{}'")
-                    .format(creds.query_tag))
-
-            connection.handle = handle
-            connection.state = 'open'
-        except snowflake.connector.errors.Error as e:
+        error = None
+        for attempt in range(1, 4):  # retry opening the connection twice after encountering an error
```

Comment:
Should we make the count of `retries` configurable, with a reasonable default, rather than hard-coding 3? Can we imagine a reason why someone might want a different number (or to disable retries entirely)?

For context:
- We'll retry getting an auth token [up to 20 times](https://github.com/dbt-labs/dbt-snowflake/pull/6/files#diff-0270ce2c5723d93d8d03173982241289438b453063e855137f8676b5451eb0a1R136)
- We make this configurable in dbt-spark via `retry_all` + `connect_retries`